routing/fseg_index.cpp

```diff
@@ -0,0 +1,39 @@
+#include "fseg_index.hpp"
+
+#include "std/utility.hpp"
+
+namespace routing
+{
+void FSegIndex::Export(vector<Joint> const & joints)
+{
+  for (JointId jointId = 0; jointId < joints.size(); ++jointId)
+  {
+    Joint const & joint = joints[jointId];
+    for (uint32_t i = 0; i < joint.GetSize(); ++i)
+    {
+      FSegId const & entry = joint.GetEntry(i);
+      RoadJointIds & roadJoints = m_roads[entry.GetFeatureId()];
+      roadJoints.AddJoint(entry.GetSegId(), jointId);
+    }
+  }
+}
+
+pair<JointId, uint32_t> FSegIndex::FindNeigbor(FSegId fseg, bool forward) const
+{
+  auto const it = m_roads.find(fseg.GetFeatureId());
+  if (it == m_roads.cend())
+    return make_pair(kInvalidJointId, 0);
+
+  RoadJointIds const & joints = it->second;
+  int32_t const step = forward ? 1 : -1;
+
+  for (uint32_t segId = fseg.GetSegId() + step; segId < joints.GetSize(); segId += step)
```

Comment:
It is works for unsigned numbers : they are modular.

I can't use any other condition: index shoud go through 3, 2, 1, 0, stops at FFF...FFF
Can't use segId >0, can't use segId >= 0, only segId < joints.GetSize()